CONTRIBUTING.md

```diff
@@ -170,6 +193,13 @@ Even if some of these dependencies are not in active use, the git "update" comma
     url = https://github.com/nhabuiduc/TypescriptSyntaxPaste.git
 ```
 
+#### Issues Compiling Proto Input
+Proto Input appears to be very sensitive to its development environment. If you are opening ProtoInput.sln in Visual Studio 2022, you may be prompted to uprade the project, but doing this might complicate things. If you cannot get it to compile, try opening a fresh version of ProtoInput.sln (so you see the ugprade prompt again) and make sure that all options are set to "No Upgrade". Upgrading can potentially lead to syntax errors in the compiling process. If you see a FILE_INFO_BY_HANDLE_CLASS syntax error pop up numerous times during compiling, this is a likely culprit.
+
+If you run into issues compiling ProtoInput.sln, there is a good chance that you are missing some dependencies or add-ons required by Visual Studio or Visual Studrio Build Tools, and you will need to identify and install them. You may also refer to the [Compilation](https://nucleus-coop.github.io/docs/compilation/) article for a more detailed example (with screenshots) of creating builds from the source code. Troubleshooting may also involve fixing ProtoInput.sln project build order dependencies, if sub-componenets appear to be building out of the intended order on your setup. In addition to basic C++ dependencies, you will also need to install C++ ATL for build tools (x86/64), at the minimum, and possibly v142 C++ dependencies. 
+
+You may also need to update the compiling order. In some dev environments, it appears that the default project compile order may result in certain subcomponents trying to compile before their dependent libraries are finished coding. Keep an eye on output errors from the compile process and note any errors that occur because a file cannot be opened. Make sure that the sub-component that failed to compile has the project that contains the file that failed to open as a dependency. To set dependencies, right-click the ProtoInput solution in the solution explorer section of Visual Studio and select the "Project Dependencies..." option. From there, simply use the drop-down box to select a subcomponent and then check the boxes next to any known, missing dependencies (as seen from output errors). Submodules with known issues in certain setups include _ProtoInputInjectorProxy_ (requires EasyHookDLL) and _ProtoInputUtilStatic_ (requies Blackbone). Check these submodules first and ensure they have their dependencies checked if if you have any issues. Otherwise use output errors to help troubleshoot any other dependency issues - a file failing to open is a good indicator that it wasn't installed yet (and is thus is a dependency that should have been installed earlier). 

```

Comment:
```suggestion
You may also need to update the compiling order. In some dev environments, it appears that the default project compile order may result in certain subcomponents trying to compile before their dependent libraries are finished coding. Keep an eye on output errors from the compile process and note any errors that occur because a file cannot be opened. Make sure that the sub-component that failed to compile has the project that contains the file that failed to open as a dependency. To set dependencies, right-click the ProtoInput solution in the solution explorer section of Visual Studio and select the "Project Dependencies..." option. From there, simply use the drop-down box to select a subcomponent and then check the boxes next to any known, missing dependencies (as seen from output errors). Submodules with known issues in certain setups include _ProtoInputInjectorProxy_ (requires EasyHookDLL) and _ProtoInputUtilStatic_ (requies Blackbone). Check these submodules first and ensure they have their dependencies checked if you have any issues. Otherwise use output errors to help troubleshoot any other dependency issues - a file failing to open is a good indicator that it wasn't installed yet (and is thus is a dependency that should have been installed earlier). 
```

Reply:
Just caught another typo in this that I'll fix in a follow-up, towards the end an extra "is" snuck in; "...(and is thus **is** a dependency that should have been installed earlier). "